### PR TITLE
Grpc gateway + buf cli on docker container to generate code

### PR DESCRIPTION
**Pull request by aabri-ankorstore**

### CODE REVIEW

File: proto/text_tools.proto [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+package text_tools;
+option go_package = "github.com/ekkinox/grpc-demo/proto";
+option php_namespace = "App\\TextTools";
+option php_metadata_namespace = "App\\GPBMetadata";
+
+import "google/api/annotations.proto";
+
+enum Transformer {
+  TRANSFORMER_UPPERCASE_UNSPECIFIED = 0;
```
Comment (ekkinox) [inline proto/text_tools.proto:11]: haha :) 

imo should be 

```
enum Transformer {
  TRANSFORMER_UNSPECIFIED = 0;
  TRANSFORMER_UPPERCASE = 1;
  TRANSFORMER_LOWERCASE = 2;
}
```

I took a shortcut in the demo and skipped the unspecified enum, but imo should be this way.

can you fix it / regen the stubs / fix the go and php code ?

File: gateway/main.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"context"
```
Comment (ekkinox) [inline gateway/main.go:4]: can you organise a bit the imports like following in the diverse go files of this PR?

```
import (
	"context"
	"net/http"
	"os"

	"github.com/rs/zerolog"
	"github.com/sirupsen/logrus"
	"github.com/ekkinox/grpc-demo/proto/github.com/ekkinox/grpc-demo/proto"
	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
	"google.golang.org/grpc"
	"google.golang.org/grpc/credentials/insecure"
)
```

I did not always do it, but it help recognize go standard modules vs vendors

**Review by ekkinox — CHANGES_REQUESTED** [review]
Amazing work !

Just the proto enum to fix / regen / code fix and we're good to go :) 

Many thx for this

File: client/main.go [hunk 3]
```diff
@@ -20,18 +20,18 @@ func main() {
...
 
-func doTransformText(c pb.TextToolsClient) {
+func doTransformText(c pb.TextToolsServiceClient) {
 
 	fmt.Println("Calling TransformText ...")
 
-	res, err := c.TransformText(context.Background(), &pb.Transformation{
-		Transformer: pb.Transformer_UPPERCASE,
+	res, err := c.TransformText(context.Background(), &pb.TransformTextRequest{
+		Transformer: pb.Transformer_TRANSFORMER_UNSPECIFIED,
```
Comment (ekkinox) [inline client/main.go:34]: UPPERCASE here

File: buf.work.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+version: v1
+directories:
+  - proto
```
Comment (ekkinox) [inline buf.work.yaml:3]: need to add a new empty line at end of new files, to avoid this

Comment (ekkinox) [reply buf.work.yaml:3]: applies to other files as well in this PR

File: proto/php/example.php [hunk 5]
```diff
@@ -25,9 +25,9 @@ public function index(string $value): Response
 
 
         [$reply, $status] = $client->TransformText(
-            new Transformation(
+            new TransformTextRequest(
                 [
-                    'transformer' => Transformer::UPPERCASE,
+                    'transformer' => Transformer::TRANSFORMER_UPPERCASE_UNSPECIFIED,
```
Comment (ekkinox) [inline proto/php/example.php:30]: did you regen the php stubs as well ? cause here we need the UPPERCASE transformer

File: server/main.go [hunk 6]
```diff
@@ -79,11 +79,11 @@ func (*server) TransformAndSplitText(stream pb.TextTools_TransformAndSplitTextSe
 	}
 }
 
-func performTransformation(t *pb.Transformation) string {
+func performTransformation(t *proto.TransformTextRequest) string {
 	switch t.Transformer {
-	case pb.Transformer_UPPERCASE:
+	case proto.Transformer_TRANSFORMER_UNSPECIFIED:
```
Comment (ekkinox) [inline server/main.go:84]: UPPERCASE here, the UNSPECIFIED like like the "null" value of he enum

**Review by ekkinox — CHANGES_REQUESTED** [review]

File: proto/php/example.php (hunk 5 above)
Comment (aabri-ankorstore) [reply proto/php/example.php:30]: sorry forget to update the php part will do in a sc

**Review by ekkinox — APPROVED** [review]

**Pull request merged**